Add useful mapping functions

MattyJ007 opened the pull request.

MattyJ007 committed:
Fix up Booleanify to not confuse users

The object mapper booleanify function makes use of the Javascript
Boolean function. The previous changes would have given unexpected
results to JS users. Rather keep existing functionality of Boolean and
indicte in the documentation the outcomse of truthy and falsy values

OHM-871

MattyJ007 committed:
Add numberify and stringify functions to OM

Useful functions for mapping.

OHM-871

BMartinos commented on test/transformationsTest.js:
```diff
@@ -237,7 +237,7 @@ test.test('Transformations', t => {
     t.test('should return null if input is null', t => {
       const input = null
-      const expected = null
+      const expected = false
```
Please update the title of the test to reflect the expected output

BMartinos commented on src/transform/primitives.js:
```diff
@@ -12,17 +12,14 @@ exports.primitives = {
+    return Number(val)
+  },
+  stringify: function(val) {
+    return String(val)
```
I wonder if we shouldnt maybe be doing `JSON.stringify` here. This way an entire object/array can be converted into a string format to be parsed again at a later stage.

MattyJ007 replied:
Sounds good :100: I'm just going to do a check for the data type coming in before doing JSON.stringify() as actual strings coming in will be returned with an extra set of inverted commas in they go through JSON.stringify().

BMartinos commented on test/transformationsTest.js:
```diff
@@ -265,4 +264,68 @@ test.test('Transformations', t => {
+
+      const result = transforms['numberify'](input)
+
+      t.equal(typeof result, 'number')
```
Can we add the type checking to the other tests as well please.

MattyJ007 replied:
I've looked at this funciton again and made some changes. Returning NaN could cause issues with JSON therefore I've changed to function to return null instead of Nan

MattyJ007 replied:
I check XML data types and it doesn't appear to accept NaN either

BMartinos replied:
Cool sounds good to me :+1:

BMartinos reviewed (changes requested):
Thanks @MattyJ007 looks good, just some minor comments

MattyJ007 committed:
Match test description to content

Boyscout

OHM-871

MattyJ007 committed:
Change OM stringify func to use JSON stringify

To handle objects being stringified. Previously arrays and objects would
not return in a useful string format

OHM-878

MattyJ007 committed:
Update numberify to return null instead of NaN

This change is to prevent errors in JSON parsing as NaN is not a valid
JSON value.

OHM-878

MattyJ007 committed:
Add more tests for function dateTimeToDate

Ensure expected behaviour is known

OHM-878

BMartinos commented on test/transformationsTest.js:
```diff
@@ -14,6 +14,26 @@ test.test('Transformations', t => {
       t.end()
     })
+
+    t.test('should return unchanged val if not in val is in unexpected format', t => {
```
I think the title of this test might is a bit confusing. `... if not in val is in ...`

MattyJ007 replied:
Fixed :100:

MattyJ007 committed:
Fix typo in test dateTimeToDate description

Boyscout

OHM-878

BMartinos reviewed (approved):
LGTM :+1:
